low-level/framework/utils/sysfs_interface.py

```diff
@@ -23,19 +23,21 @@
 from pathlib import Path
 
 from framework.utils.utility import Utility
+from framework.utils.conf_utils import SSPL_CONF, Conf
 
 class SysFS(Utility):
     """Module which is responsible for fetching information
        internal to system such as SAS Port/SAS phy/network
        using /sys file system"""
 
-    sysfs = "/sys/class/"
-    cpu_online_fp = "/sys/devices/system/cpu/online"
     nw_phy_link_state = {'0':'DOWN', '1':'UP', 'unknown':'UNKNOWN'}
 
     def __init__(self):
         """init method"""
         super(SysFS, self).__init__()
+        self.sysfs_base_path = self.get_sysfs_base_path()
```

Comment:
As told in scrum, we need to handle error in consuming sensors if sysfs base dir doesnt exist. And shutdown that sensor with an error log

Reply:
This flow is already in place.
For eg:- In CPUFaultSensor,

>     try:
            self.current_cpu_info = self._utility_instance.get_cpu_info()
        except Exception as e:
            logger.error(f"Error while reading current cpu info, shutting down CPUFaultSensor : {e}")
            self.shutdown()
 
get_cpu_info internally reads the </sys/devices/..> file and returns the status.
Any error will be catched and sensor shutdown will be called.
same for other sensors as well.

@cdeshmukh still require other handling?

Reply:
yes... same is the scenario for other sensors as well.

Reply:
good